x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -789,6 +789,26 @@ public void testRankWithFiltersPushdown() {
         assertThat(actualLuceneQuery.toString(), is(expectedLuceneQuery.toString()));
     }
 
+    public void testSearchWithMatchFilters() {
+        var plan = plannerOptimizer.plan("""
+            search test [
+                | WHERE match(first_name, "Meg")
+                | WHERE match(last_name, "Ryan")
+            ]
+            """);
+
+        var limit = as(plan, LimitExec.class);
+        var exchange = as(limit.child(), ExchangeExec.class);
+        var project = as(exchange.child(), ProjectExec.class);
+        var fieldExtract = as(project.child(), FieldExtractExec.class);
+        var actualLuceneQuery = as(fieldExtract.child(), EsQueryExec.class).query();
+
+        var expectedLuceneQuery = new BoolQueryBuilder().filter(
+            new BoolQueryBuilder().must(new MatchQueryBuilder("first_name", "Meg")).must(new MatchQueryBuilder("last_name", "Ryan"))
```

Comment:
this is an interesting behaviour that we can follow up on separately.
I would expect that we just have a bool query with two filter clauses:
```
 var expectedLuceneQuery =  new BoolQueryBuilder().filter(new MatchQueryBuilder("first_name", "Meg")).filter(new MatchQueryBuilder("last_name", "Ryan"))
```
instead we are using `must` clauses which comes from how the `PushFiltersToSource` forms the query, specifically here:

https://github.com/elastic/elasticsearch/blob/90047f94d2dcdf54731918b2810d3e4550943c07/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java#L222

We can see that this behaviour happens with `FROM` too, take this test query:

https://github.com/elastic/elasticsearch/blob/dabd9b62516569a783bbb1aa99b9de618c1d5f4b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java#L236-L240

the filters are pushed as `must` clauses:

https://github.com/elastic/elasticsearch/blob/dabd9b62516569a783bbb1aa99b9de618c1d5f4b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java#L255

I think maybe they should be pushed as `filter` instead since `must` clauses add to the score which for filters we don't care about?

Reply:
That's a good question.  It works ok, by I also suspect that `filter` would be more appropriate here.